quack_app/lib/screens/login/login_screen.dart

```diff
@@ -13,6 +13,7 @@ class LoginScreen extends StatefulWidget {
 }
 
 class _LoginScreenState extends State<LoginScreen> {
+  late MenuData menuData;
```

Comment:
Don't instantiate it, use it as a Singleton: `MenuData().something`. That's actually what you're doing -- no need to define it.